source/libraries/Connector.ts

```diff
@@ -1,35 +1,24 @@
-import * as getPort from "get-port";
 import EthjsHttpProvider = require('ethjs-provider-http');
 import EthjsQuery = require('ethjs-query');
+import { Configuration } from './Configuration';
 
 export class Connector {
-    private ethjsQuery: EthjsQuery|null = null;
+    public readonly ethjsQuery: EthjsQuery;
 
-    public async getEthjsQuery(): Promise<EthjsQuery> {
-        if (this.ethjsQuery === null) {
-            this.ethjsQuery = await this.connect();
-        }
-        return this.ethjsQuery;
+    constructor(configuration: Configuration) {
+        const ethjsHttpProvider = new EthjsHttpProvider(`http://${configuration.httpProviderHost}:${configuration.httpProviderPort}`);
+        this.ethjsQuery = new EthjsQuery(ethjsHttpProvider);
     }
 
     public async waitUntilConnected(): Promise<EthjsQuery> {
-        const ethjsQuery = await this.getEthjsQuery();
         while (true) {
             try {
-                await ethjsQuery.net_version();
+                await this.ethjsQuery.net_version();
                 break;
             } catch {
                 // swallow the error, we just want to loop until the net_version call above succeeds.
```

Comment:
I'd say check for the specific error you're expecting to swallow here, too likely that something upstream breaks and we end up swallowing an error that we're not expecting.

Reply:
Normally I would agree with you on this, but the problem with that strategy in this case is that there is a whole slew of possible errors you might get back and I'm not confident I could reliably enumerate them all.  Depending on the current state of the node at the time the request happens to land, and the node you are talking to (Geth, Parity, TestRPC), you could get anything from a socket timeout, TCP connection error, JSON-RPC error, or any other kind of transport/protocol error at any of many different layers of the stack.  `this.ethjsQuery.net_version` is picked as a very simple method that really has no failure conditions other than "remote side not on or functioning correctly", so any error we get back is indicative of either a programming error (e.g., `this.ethjsQuery` was constructed with a bad host/port) or the remote end is not on or in a bad state or is not an Ethereum node.